lib/python/qmk/cli/new/keyboard.py

```diff
@@ -0,0 +1,182 @@
+        ( 'atmega32a'  , 'ps2avrgb' ),
+        ( 'atmega32u2' , 'avr' ),
+        ( 'atmega32u4' , 'avr' ),
+        ( 'STM32F042'  , 'stm32' ),
+        ( 'STM32F072'  , 'stm32' ),
+        ( 'STM32F103'  , 'stm32' ),
+        ( 'STM32F303'  , 'stm32' ),
+    )
+
+    # Generate the USB Product ID
+    def generate_pid(str):
+        str = str.encode('utf-8')
+        str = hashlib.sha1(str).hexdigest()[0:4].upper()
+        return str
+
+    # Rewrites the %YEAR%, %YOUR_NAME%, %KEYBOARD% and %PID% placeholders in the
+    #   new files.
+    def rewrite_source(file):
+        rw_file = Path(keyboard_path) / file
+        file_contents = rw_file.read_text() \
+            .replace("%YEAR%", str(year)) \
+            .replace("%YOUR_NAME%", user_name) \
+            .replace("%KEYBOARD%", keyboard) \
+            .replace("%FINAL_DIR%", final_directory) \
+            .replace("%KEYBOARD_NAME%", keyboard_name) \
+            .replace("%PID%", pid) \
+            .replace("%MCU%", mcu)
+        rw_file.write_text(file_contents)
+
+    def print_microcontrollers():
+        for i, mcu in enumerate(mcus, 1):
+            cli.echo("     %s: %s (%s)", str(i).rjust(2, " "), mcu[0], mcu[1] )
+
+
+    if cli.args.keyboard:
+        keyboard = keyboard_name = cli.args.keyboard
+    else:
+        # ask for user input if keyboard was not provided in the command line
+        print("""\n** What is the project name? **
+
+        This will be the name used to compile firmware for your keyboard. The
+        project name must match the regular expression `^[a-z0-9][a-z0-9_]+`.
+
+        Files will be placed in `qmk_firmware/keyboards/<project_name>/`.""")
+        keyboard = input("\n    Project Name: ")
+        keyboard = re.sub(r'[^a-z0-9_]', "", keyboard.lower())
+
+
+    if cli.args.project:
+        keyboard_name = cli.args.project
+    else:
+        # Ask the common name of the keyboard, if not specified by --project
+        print("""\n** What is the keyboard's name? **
+
+        This is the name that people will use to refer to your keyboard. It should
+        be something human-friendly, like \"Clueboard 66%\" or \"Ergodox EZ\".""")
+        keyboard_name = input("\n    Name: ")
+
+
+    if cli.args.microcontroller:
+        mcu = cli.args.microcontroller.lower()
+        for key, value in enumerate(mcus):
+            if re.match( r'^' + value[0] + '$', mcu, re.I ):
+                arch = value[1]
+                break
+        else:
+            cli.log.error(mcu + " is not a valid microcontroller option.")
+            options = [];
+            for i in range(0, len(mcus)):
+                options.append( mcus[i][0] )
+            print("  Valid Options: ", ", ".join(options) )
+            exit(1)
+    else:
+    # Ask what microcontroller is being used
+        print("\n** Select the microcontroller used: **\n")
+        print_microcontrollers()
+
+        mcu = int(input("\n    Microcontroller: (1-" + str(len(mcus)) + "): "))
```

Comment:
```suggestion
        mcu = question("Microcontroller: (1-%d): ", answer_type=int, len(mcus))
```

Reply:
Was unable to get this method to work. Even on a rebased branch from current master, using this method causes the script to fail to run.

Reply:
`mcu = question("Microcontroller: (1-%d): ", answer_type=int, len(mcus))`

Reply:
Oh, so _that's_ what that error message means :bulb: